mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -541,3 +546,12 @@ case class ScalaPresentationCompiler(
     }
   }
 }
+
+object ScalaPresentationCompiler {
+  private[this] val deprecatedTagList: List[CompletionItemTag] =
+    List(CompletionItemTag.Deprecated)
+
+  /** Add the `Deprecated` tag to the completion item. */
+  private def setCompletionItemDeprecated(value: CompletionItem): Unit =
+    value.setTags(deprecatedTagList.asJava)
```

Comment:
A couple thoughts on this. We also use tags to tag deprecated in the `CompletionProvider` as we can see here https://github.com/scalameta/metals/blob/06f59123c61eb6124f6cb98762de3748d4e7a04e/mtags/src/main/scala-2/scala/meta/internal/pc/CompletionProvider.scala#L227-L229.

So if we were to stick with having a utility like this, it might be best to use it both places. On the other hand, if we ever decide to use more tags for whatever reason, this sort of falls apart since this doesn't append, just replaces. So I guess I'd prefer one of the following:

- Change this to instead of just setting tags to deprecated, take in ca completion item, check if there are any other tags, concat and set them, and then also use it here and in the `CompletionProvider`
- Simply not worry about having this here now, and inline it like we do in `CompletionProvider`. This is the easier option, and probably the route I'd go.

What do you think?

Reply:
I'm 👍 on inlining this too. Despite being a little verbose, I don't think it's worth the indirection

Reply:
I didn't consider that it would be overwriting other tags, that's a valid point. 

It may be worth noting that as of now there is only a single possible tag, `Deprecated`. Though I assume the reason lsp4j is changing this API is that they anticipate adding more soon.

With that in mind, I think something like this would be more appropriate,

```scala
      (item.getTags.asScala.toSet ++ Set(CompletionItemTag.Deprecated)).toList.asJava
```

This assumes that having more than one tag of the same type is semantically the same, e.g. `List(Deprecated)` means the same as `List(Deprecated, Deprecated)`. Whether or not that assumption holds true depends on what other tags lsp4j adds, but I think it is _likely_ it will hold true.

For my personal tastes, that is quite a lot of Java<->Scala back and forth for inlining, but if that is preferred route to go I'll update the PR to that.

Reply:
Updated